rust/gui-client/src-tauri/src/client/network_changes.rs

```diff
@@ -1,355 +1,9 @@
-//! A module for getting callbacks from Windows when we gain / lose Internet connectivity
-//!
-//! # Latency
-//!
-//! 2 or 3 seconds for the user clicking "Connect" or "Disconnect" on Wi-Fi,
-//! or for plugging or unplugging an Ethernet cable.
-//!
-//! Plugging in Ethernet may take longer since it waits on DHCP.
-//! Connecting to Wi-Fi usually notifies while Windows is showing the progress bar
-//! in the Wi-Fi menu.
-//!
-//! DNS server changes are (TODO <https://github.com/firezone/firezone/issues/3343>)
-//!
-//! # Worker thread
-//!
-//! `Listener` must live in a worker thread if used from Tauri.
-//! `
-//! This is because both `Listener` and some feature in Tauri (maybe drag-and-drop) depend on COM.
-//! `Listener` works fine if we initialize COM with COINIT_MULTITHREADED, but
-//! Tauri initializes COM some other way.
-//!
-//! In the debug command we don't need a worker thread because we're the only code
-//! in the process using COM.
-//!
-//! I tried disabling file drag-and-drop in tauri.conf.json, that didn't work:
-//! - <https://github.com/tauri-apps/tauri/commit/e0e49d87a200d3681d39ff2dd80bee5d408c943e>
-//! - <https://tauri.app/v1/api/js/window/#filedropenabled>
-//! - <https://tauri.app/v1/api/config/#windowconfig>
-//!
-//! There is some explanation of the COM threading stuff in MSDN here:
-//! - <https://learn.microsoft.com/en-us/windows/win32/api/objbase/ne-objbase-coinit>
-//! - <https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-coinitializeex>
-//!
-//! Raymond Chen also explains it on his blog: <https://devblogs.microsoft.com/oldnewthing/20191125-00/?p=103135>
+#[cfg(target_os = "linux")]
+#[path = "network_changes/linux.rs"]
+mod imp;
 
-use anyhow::Result;
-use std::sync::Arc;
-use tokio::{runtime::Runtime, sync::Notify};
-use windows::{
-    core::{ComInterface, Result as WinResult, GUID},
-    Win32::{
-        Networking::NetworkListManager::{
-            INetworkEvents, INetworkEvents_Impl, INetworkListManager, NetworkListManager,
-            NLM_CONNECTIVITY, NLM_NETWORK_PROPERTY_CHANGE,
-        },
-        System::Com,
-    },
-};
+#[cfg(target_os = "windows")]
+#[path = "network_changes/windows.rs"]
+mod imp;
```

Comment:
what does `imp` stand for?

Reply:
"Implementation". I'm open to better names, I just keep waffling between "impl" "detail" and "inner" for these things where I can't think of anything descriptive